internal/runners/languages/install.go

```diff
@@ -53,10 +51,7 @@ func (u *Update) Run(params *UpdateParams) error {
 
 func parseLanguage(langName string) (*model.Language, error) {
 	if !strings.Contains(langName, "@") {
-		return &model.Language{
-			Name:    langName,
-			Version: "",
-		}, nil
+		return nil, locale.NewInputError("err_languages_parameter", "Invalid parameter for language argument, must be of the form <language>@<version>")
```

Comment:
`@version` is still optional. Remember the point of this story is consistent behaviour with `state install`, in which it is also optional.

Reply:
I think what you're touching on here is the difference in behaviour when installing vs when updating. When updating you should supply a specific version. But when you're just installing it should just detect the version if not supplied.

I think we should probably just say "X is already installed" if you run `install ..` without specifying a version. @rawktron thoughts?

Reply:
It seems like `state install python` should say `python<version> is already installed` if run in a python project. If run in a project with a different language it will replace it with the latest python?

And then `state install python@version` will update the version of python if in a python project. Otherwise it will install(replace) with that specific version.

Just want to confirm what the desired behaviour is before I make the changes.

Reply:
Yes that sounds right to me. But for the first one it should also suggest that they can update, eg.

```
python is already installed, you can update it by running python@<version>
```

Reply:
The dashboard does not allow switching the language of a project, and we already have an existing check for this. I have left that part as is.

Now `state install python` in a python project will return the error you suggested @Naatan. 
`state install python@<version>` will install the version requested.
`state install python` in a Perl project will return an error stating that you cannot switch the language of a project.